dispatch

```diff
@@ -1,44 +1,42 @@
 #!/bin/bash
-# This hook installs the centos dependencies needed to run the charm,
+# This hook installs the dependencies needed to run the charm,
 # creates the dispatch executable, regenerates the symlinks for start and
 # upgrade-charm, and kicks off the operator framework.
 
 set -e
 
-# Source the os-release information into the env.
+# Source the os-release information into the env
 . /etc/os-release
 
 if ! [[ -f '.installed' ]]
 then
-	# Determine if we are running in centos or ubuntu, if centos
-	# provision the needed prereqs.
-	if [[ $ID == 'ubuntu' ]]
-	then
-		echo "Running Ubuntu."
-		# necessary to compile and install NHC
-		apt-get install --assume-yes make automake
-	elif [[ $ID == 'centos' ]]
+	if [[ $ID == 'centos' ]]
 	then
+		# Necessary to compile and install NHC
+		yum -y install make
+
 		# Determine the centos version and install prereqs accordingly
-		major=$(cat /etc/centos-release | tr -dc '0-9.'|cut -d \. -f1)
-			echo "Running CentOS$major, installing prereqs."
-		if [[ $major == "7" ]]
+		if [[ $VERSION_ID == "7" ]]
 		then
-			yum -y install epel-release
-			yum -y install yum-priorities python3 make automake yum-utils
-		elif [[ $major == "8" ]]
+			/bin/bash ./src/scripts/install_python.sh
+		elif [[ $VERSION_ID == "9" ]]
 		then
-			dnf -y install epel-release
-			dnf -y install yum-priorities python3 make automake yum-utils
-		else
-			echo "Running unsuppored version of centos: $major"
-			exit -1
+			yum -y install python38
 		fi
-	else
-		echo "Running unsuppored os: $ID"
-		exit -1
+	elif [[ $ID == 'ubuntu' ]]
+	then
+		# Necessary to compile and install NHC
+		apt-get install --assume-yes make
 	fi
 	touch .installed
 fi
 
-JUJU_DISPATCH_PATH="${JUJU_DISPATCH_PATH:-$0}" PYTHONPATH=lib:venv ./src/charm.py
+# set the correct python bin path
+if [[ $ID == "centos" && $VERSION_ID == "7" ]]
+then
+	PYTHON_BIN=/opt/python/3.8.16/bin/python3.8
```

Comment:
Could change this to `/usr/bin/env python3.8`. This way we don't need to hardcode the PATH and can let env find the Python 3.8 interpreter for us.

Reply:
Done!